tests/page/expect-matcher-result.spec.ts

```diff
@@ -0,0 +1,92 @@
+/**
+ * Copyright (c) Microsoft Corporation.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { stripAnsi } from '../config/utils';
+import { test, expect } from './pageTest';
+
+test('toMatchText-based assertions should have matcher result', async ({ page }) => {
+  await page.setContent('<div id=node>Text content</div>');
+  const locator = page.locator('#node');
+
+  const e1 = await expect(locator).toHaveText(/Text2/, { timeout: 1 }).catch(e => e);
+  e1.matcherResult.message = stripAnsi(e1.matcherResult.message);
+  expect(e1.matcherResult).toEqual({
+    actual: 'Text content',
+    expected: /Text2/,
+    message: expect.stringContaining(`Timed out 1ms waiting for expect(locator('#node')).toHaveText(/Text2/)`),
+    name: 'toHaveText',
+    pass: false,
+  });
+
+  const e2 = await expect(locator).not.toHaveText(/Text/, { timeout: 1 }).catch(e => e);
+  e2.matcherResult.message = stripAnsi(e2.matcherResult.message);
+  expect(e2.matcherResult).toEqual({
+    actual: 'Text content',
+    expected: /Text/,
+    message: expect.stringContaining(`Timed out 1ms waiting for expect(locator('#node')).not.toHaveText(/Text/)`),
+    name: 'toHaveText',
+    pass: true,
```

Comment:
How could `pass` be true? I find this API to be pretty strange for our users.

Reply:
That is not for us to decide, failing not will have pass set to true in expect.